changelog/dmd.source-archive.dd

```diff
@@ -0,0 +1,107 @@
+# Support Source Archive Files
+
+This is a compiler feature, not a D language feature.
+
+Similar to how libraries of object files are made available to the linker,
+this adds source archive file support to the compiler. Any package (and all its
+sub-files) can become a source archive file. The source archive file is then
+supplied to the compiler rather than a directory with lots of files in it.
+
+This means, for example, that all of Phobos can be distributed as a single
+file, std.sar. (The .sar extension stands for "source archive".) If std.sar
+is in a path along the import path list supplied to the compiler, the
+compiler will prefer std.sar to looking through the std directory tree for the
+sub-modules. The std directory wouldn't even need to exist.
```

Comment:
What happens to debug info if this was the case?

Reply:
How would gdb let me step through phobos functions?

Reply:
Unless the debugger supported .sar files, they won't be able to use them. In such cases, the user will need to expand the archive back into individual files. This would be true for all tools we don't have control over.

For example, having one's code editor support them.

This would be come worthwhile if .sar files catch on.

Reply:
Why does debug info matter? We're shipping the source code, not object files.

Reply:
The debug info points to the code, which then wouldn't actually exist on the disk. This would apply to both (say) a phobos dso and templates (whose implementation would be in the importers object file, but debug info would point to the again missing code in the .sar)

Reply:
> In such cases, the user will need to expand the archive back into individual files. 

The issue with that is that the one place where I could see (lets say it nets a 5% win) this actually being used is  the standard library and druntime, where debuggers do actually work (i.e. can let you step through the code in VisualD and so on), but if there was another step in that process (a manual one) you can end up throwing away a lot of that (which is admittedly held together with bits of string but its bit of string that are also holding D's adoption together frankly).